src/app/pages/MarginTradePage/components/TradingHistory/utils.tsx

```diff
@@ -0,0 +1,337 @@
+import React, { useState } from 'react';
+import { useSortBy, useTable } from 'react-table';
+import { useTranslation } from 'react-i18next';
+import { translations } from 'locales/i18n';
+import { bignumber } from 'mathjs';
+import { Tooltip } from '@blueprintjs/core';
+import { Asset } from '../../../../../types';
+import { TradingPosition } from 'types/trading-position';
+import { AssetsDictionary } from '../../../../../utils/dictionaries/assets-dictionary';
+import { TradingPairDictionary } from 'utils/dictionaries/trading-pair-dictionary';
+import { toWei, weiTo18 } from 'utils/blockchain/math-helpers';
+import { TradeProfit } from 'app/components/TradeProfit';
+import { PositionBlock } from '../OpenPositionsTable/PositionBlock';
+import { LinkToExplorer } from '../../../../components/LinkToExplorer';
+import { weiToNumberFormat } from '../../../../../utils/display-text/format';
+import { Pagination } from '../../../../components/Pagination';
+
+export type EventType = 'buy' | 'sell';
+
+export interface ICustomEvent {
+  loanId: string;
+  loanToken: Asset;
+  collateralToken: Asset;
+  position: TradingPosition;
+  type: EventType;
+  leverage: string;
+  positionSize: string;
+  price: string;
+  txHash: string;
+  isOpen: boolean;
+  time: number;
+  event: string;
+  collateralToLoanRate: string;
+}
+
+export interface IEventData {
+  data: Array<ICustomEvent>;
+  isOpen: boolean;
+  loanId: string;
+}
+
+export interface ICalculatedEvent {
+  loanId: string;
+  position: TradingPosition;
+  loanToken: Asset;
+  collateralToken: Asset;
+  leverage: string;
+  positionSize: string;
+  entryPrice: string;
+  closePrice: string;
+  profit: string;
+  entryTxHash: string;
+  closeTxHash: string;
+  time: number;
+}
+
+export const HistoryTable = (props: { items: ICalculatedEvent[] }) => {
```

Comment:
`React.FC`

Reply:
Also, this component shouldn't be here at all, this is `utils` file. Move it to the appropriate location.

Reply:
done